### DISCUSSION TITLE
How Google detect AMP page if we use "hybrid" ?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#20570] [Help] [answered]
Hi, I am trying make site fastest on mobile. But how can Google detect link `/post/my-post?amp=1 ` ???
Is bot smart enough to find it ? Or I must add the link with query amp=1 to sitemap.xml ? 
Or will we rediect user to link `/post/my-post?amp=1 ` if they are on mobile ?

**Comment by Dragate** [comment, marked as answer]
From what I've seen, declaring a page as "hybrid" makes Next.js add to the head of the non-AMP page a link to the AMP page (e.g. `<link rel="amphtml" href="/post/my-post?amp=1">`). So, when a search engine crawls your page it knows that there's an AMP variant of the page and stores it in the AMP cache to serve it to mobile users. You don't need to add it to the sitemap.

**Reply by IRediTOTO** [reply]
lets me try, thanh you. I want >=90 point, so hard...

**Reply by IRediTOTO** [reply]
@Dragate  I have another question. When user click to search result, if they use desktop they will have norman page, right ? Or I must redirect page ? @@

**Reply by Dragate** [reply]
Yea, as far as I know, the search engine will serve the AMP page only to mobile users (at least that's what I see Google do). The rest of the users will get the normal page, as usual. You don't need to do any redirect.

**Reply by lachlanjc** [reply]
Mark this as the answer @IRediTOTO? :+1:
